File: JavaScript Front End/src/Messages/CommandResponseMessages/GraphBuildingTaskCreationSuccessResponseMsg.js
const InfoSecCommandResponseMsg = require('./InfoSecCommandResponseMsg.js').InfoSecCommandResponseMsg;

class GraphBuildingTaskCreationSuccessResponseMsg extends InfoSecCommandResponseMsg
{
    constructor(jsonMessage)
    {
        super(jsonMessage);
        this.createdNodeId = this.jsonMessage["createdNodeId"];
        this.createdNodeTemporaryId = this.jsonMessage["createdNodeTemporaryId"];
    }
}

module.exports = {GraphBuildingTaskCreationSuccessResponseMsg};

File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandResponseMessages/GraphBuildingTaskCreationSuccessResponseMsg.java
package InfoSecCooker.Communication.Messages.CommandResponseMessages;

public class GraphBuildingTaskCreationSuccessResponseMsg extends GraphBuildingSuccessResponseMsg
{
    private long createdNodeId;
    private long createdNodeTemporaryId;

    public GraphBuildingTaskCreationSuccessResponseMsg(long createdNodeId, long createdNodeTemporaryId, String additionalInfo)
    {
        super(additionalInfo);
        this.createdNodeId = createdNodeId;
        this.createdNodeTemporaryId = createdNodeTemporaryId;
    }

    public GraphBuildingTaskCreationSuccessResponseMsg(long createdNodeId, long createdNodeTemporaryId)
    {
        super("");
        this.createdNodeId = createdNodeId;
        this.createdNodeTemporaryId = createdNodeTemporaryId;
    }
}
